### PR TITLE
[enhancement] AutoEat food preference.

### LINKED ISSUE #1853: AutoEat better preference of food
if health goes below x > eat gap
if hunger goes below x > eat food

if on fire and have an enchanted gap -> eat it

### PR DESCRIPTION
**Pull request by scorbett123**
if health goes below x > eat gap
if hunger goes below x > eat food

if on fire and have an enchanted gap -> eat it

closes #1853

### CODE REVIEW

File: src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt [hunk 1]
```diff
@@ -143,25 +173,32 @@ internal object AutoEat : Module(
...
+        if ((item !is ItemFood) || (item == Items.CHORUS_FRUIT) || (isBadFood(itemStack, item) and !eatBadFood)) {
+            return false
+        }
 
-        return item is ItemFood
-            && item != Items.CHORUS_FRUIT
-            && (eatBadFood || !isBadFood(itemStack, item))
-            && player.canEat(item == Items.GOLDEN_APPLE)
+        return prefferedLevel.isValid(item, itemStack)
     }
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt:202]: I don't see isValid being used anywhere?? Am I missing something?

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt [hunk 2]
```diff
@@ -26,13 +24,36 @@ internal object AutoEat : Module(
...
+    private val eGapOnFire by setting("EGap on fire", true)
     private val eatBadFood by setting("Eat Bad Food", false)
     private val pauseBaritone by setting("Pause Baritone", true)
 
     private var lastSlot = -1
     private var eating = false
 
+    private var prefferedLevel = PreferedLevel.NORMAL
+
+    enum class PreferedLevel {
```
[the remark this answers is not in the document]
Comment (Xiaro) [reply src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt:37]: Why not just call it preferred food?

[the remark this answers is not in the document]
Comment (l1ving) [reply src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt:37]: That works too. I was more about correcting the spelling

**Comment by scorbett123** [comment]
Sorry for my dodgy spelling.

File: src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt (hunk 1 above)
Comment (scorbett123) [reply src/main/kotlin/org/kamiblue/client/module/modules/player/AutoEat.kt:202]: It is, line 187, 88, 85.

**Comment by Xiaro** [comment]
Thanks

**Comment by scorbett123** [comment]
Thanks

**Pull request merged**